nanome/_internal/_process/_process_manager.py

```diff
@@ -183,18 +183,18 @@ def _remove_session_processes(self, session_id):
             self.__pending.remove(entry)
 
         for entry in running:
-            entry.process.kill()
+            entry.process.terminate()
```

Comment:
```suggestion
            entry.process.kill()
```
this was intentionally `kill` as `SIGTERM` can be ignored by the process, whereas `SIGKILL` is on the kernel level and can't be blocked.

Reply:
I wasn't sure if it needed to be kill, that was the first command I found. I saw that we actually use terminate in the file already, and it seems better to gracefully close everything.

Reply:
right, `terminate` is indeed better for gracefully closing a process, but if a process is frozen or just programmed to ignore `SIGTERM`, `kill` is required. I guess the ideal solution would be to `terminate` first, then `kill` after x seconds if the process is still running